sql/tests: TestRandomSyntaxSQLSmith failed [expected equivalence determinant to be single col]

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17419822?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17419822?buildTab=artifacts#/) on release-24.1 @ [fc14401b759abf335bb2442a8995112817a14027](https://github.com/cockroachdb/cockroach/commits/fc14401b759abf335bb2442a8995112817a14027):

Random syntax error:

```
    rsg_test.go:960: Crash detected: server panic: pq: internal error: expected equivalence determinant to be single col: ()-->(43,46,47), (42,45)==(44), (44)==(42,45) (1)
```
Query:

```
        WITH
        	with_146356 ("c ol_848159", col_848160, "col _848161", " col_848162", "c	oL_848163", "col%v_848164")
        		AS (
        			SELECT
        				tab_356875.tableoid AS "c ol_848159",
        				'':::VOID AS col_848160,
        				COALESCE(NULL, tab_356875.tableoid) AS "col _848161",
        				tab_356875.tableoid AS " col_848162",
        				B'01001010011101111011111101' AS "c	oL_848163",
        				'zmv':::rand_typ_1 AS "col%v_848164"
        			FROM
        				defaultdb.public.seed@seed__int8__float8__date_idx AS tab_356875
        		),
        	with_146357 ("col\n_848165")
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						('UZ-T':::STRING),
        						(e'\x161':::STRING),
        						(e'5s\x16[':::STRING),
        						(e'p\x15\'':::STRING),
        						(e'G0n\x0e#%%Ae':::STRING),
        						('M+':::STRING)
        				)
        					AS tab_356876 ("col\n_848165")
        		)
        SELECT
        	cte_ref_42268."col%v_848164" AS col_848166,
        	3946663984:::OID AS col_848167,
        	tab_356877.col1_7 AS "c.ol_848168",
        	tab_356877.col1_0 AS "-col_848169"
        FROM
        	with_146357 AS "cte	_re'f_42267", defaultdb.public.table_1@[0] AS tab_356877, with_146356 AS cte_ref_42268
        WHERE
        	false
        ORDER BY
        	"cte	_re'f_42267"."col\n_848165" DESC NULLS LAST
        LIMIT
        	68:::INT8;
```
Schema:

```
    rsg_test.go:763: To reproduce, use schema:
    rsg_test.go:765: SET sql_safe_updates = false;;
    rsg_test.go:765: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:765: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:765: CREATE TABLE table_1 (col1_0 TIME NOT NULL, "
        col1 _1" VARCHAR NOT NULL, col1_2 GEOGRAPHY NOT NULL, "col 1_3" BIT(7) NOT NULL, "co%qL1_4" OID NULL, co😬l1_5 BIT(48) NOT NULL, "cOl1_6" STRING NOT NULL AS (lower(CAST(col1_2 AS STRING))) VIRTUAL, col1_7 STRING NOT NULL AS (lower(CAST(col1_0 AS STRING))) STORED, INDEX (lower(CAST("col 1_3" AS STRING)) DESC) STORING ("
        col1 _1", "co%qL1_4"), UNIQUE (lower(CAST(co😬l1_5 AS STRING)) ASC, co😬l1_5 DESC, "col 1_3" DESC) WHERE (((table_1."
        col1 _1" >= 'X':::STRING) OR (table_1."cOl1_6" != e'\U00002603':::STRING)) AND (table_1.col1_7 != 'X':::STRING)) OR (table_1.col1_0 < '24:00:00':::TIME), INVERTED INDEX ("
        col1 _1" DESC, col1_0, "cOl1_6" ASC, co😬l1_5, col1_7 gin_trgm_ops ASC), INDEX (col1_7 DESC) STORING (col1_2) PARTITION BY LIST (col1_7) (PARTITION table_1_part_0 VALUES IN (('T':::STRING,), (e'\x07BM':::STRING,), ('"':::STRING,), (e'Ld\x12':::STRING,), (e'CS\x03':::STRING,))), INDEX ("co%qL1_4", co😬l1_5, "col 1_3") PARTITION BY LIST ("co%qL1_4", co😬l1_5, "col 1_3") (PARTITION table_́1_part_0 VALUES IN ((0:::OID, B'100100011011001101110110011100110100001101100110', B'1000001'), (175801156:::OID, B'010000011110011101000100101011100101101101100010', B'0101101'), (2650093818:::OID, B'011111111111111111111111111111111111111111111111', B'0111001'), (738436528:::OID, B'011000100011000100011100010101100110101110010000', B'1001100'), (1972333496:::OID, B'010001110110101100110101011001000101000011110001', B'1111010'), (2182371432:::OID, B'111111001100001110100100100001100110100111010000', B'1011111'), (3155402811:::OID, B'100010010100001101111110001011100001101010010011', B'0111111'), (128604798:::OID, B'000100000001011110001101001010000101100000000111', B'1111111'), (3903677121:::OID, B'010001110001010110100101011110011101001010111001', B'1011000'), (0:::OID, B'010010001001000001000001100010001011010111000111', B'0111000')), PARTITION "tab le_1_part_1" VALUES IN ((2274133519:::OID, B'111001100001010101110110110101110100000100010101', B'0100011'), (82999706:::OID, B'110010000010001001101100101000110101011001111100', B'0001101'), (1451023144:::OID, B'001100000001101010101110100011001011011001011110', B'0001110'), (3467615690:::OID, B'010111110110111000111110100110100101011100011011', B'1001111'), (1864058659:::OID, B'100110010001100001001001101101000000100111110111', B'0101100'), (520762683:::OID, B'010101111110010000101110001111011010011111001001', B'1000001'), (3155122434:::OID, B'000011101010001110100110000011000000011010010000', B'1100000'), (2359974287:::OID, B'111111111101101000001101010000111110110011100000', B'0111001'), (0:::OID, B'000001011000001101110100100100111010101100010001', B'0111111'), (1538411305:::OID, B'101010100111101011110001101100010101010110010011', B'1100110')), PARTITION table_1_part_2 VALUES IN ((1294885822:::OID, B'001101010010100010011001100110111011110110111000', B'1110110'), (1693652289:::OID, B'110001101000101011100101000000111000111100110111', B'1000111'), (2521324317:::OID, B'011100000000010011111110101101010110001100010011', B'1100001'), (1617204781:::OID, B'011100000000111000101000100111010011000111011101', B'1001000'), (3907366815:::OID, B'010101010111111010100010011110001101000111100110', B'1111111'), (3034942998:::OID, B'111110101110100000111000111011001100111101011101', B'1101000'), (541265005:::OID, B'000001001011011011001110101100111100110011011110', B'0111111'), (4222616567:::OID, B'011110000001001101000001111110111111111011011111', B'1001000'), (1950934094:::OID, B'000001100001100111100111101011111110101110010100', B'1101000'), (3730028164:::OID, B'101111110101100001000110011110111011101110010110', B'1010111')), PARTITION table_1_part_3 VALUES IN ((3552933082:::OID, B'110011001011010100001001001000101001010110001001', B'0001110'), (422102962:::OID, B'111000111011100000001110100001100100110101011001', B'1110110'), (0:::OID, B'011001111010010011011011101101111100010011110100', B'1010101'), (1944753059:::OID, B'100100100111100000010000110110101100101111111001', B'0000000'), (0:::OID, B'110111101100101110000001001001100100101000101100', B'0101011'), (2066092950:::OID, B'101100110111010000000011111111111000111110111111', B'0000000'), (3443594400:::OID, B'100111011100000010101001000101101011101010000100', B'0111111'), (1701110700:::OID, B'001110010101101010110010011010111101100000101010', B'1000000'), (3674080252:::OID, B'111001010101100101110110101101001111101100110001', B'1101011'), (2496517517:::OID, B'001011001000001110000010100010010001100100100011', B'0111111')), PARTITION table_1_part_4 VALUES IN ((298697776:::OID, B'001000111111110111110010011001101101111010001000', B'0111011'), (0:::OID, B'110001110100110101110110111010000001101101110010', B'1110001'), (3689463023:::OID, B'110110000111000000101111100100100111001110001000', B'0111111'), (720881842:::OID, B'101110001110101001011011101111111010000111011110', B'0100100'), (1610972851:::OID, B'010010001111111010000111001100101101110111101010', B'0001111'), (1486618566:::OID, B'001101000110011111010110010111010001010001011010', B'1010110'), (3017486610:::OID, B'111010011111000111010100110011010011010001011110', B'1101110'), (747664885:::OID, B'001011010101100100010000101101101100010101100000', B'0010100'), (4221155320:::OID, B'011110100001111111101000101011101010011111010110', B'0100101'), (3182658585:::OID, B'111100001001011011110011010100011101001110101010', B'1001111'))));
    rsg_test.go:765: ALTER TABLE table_1 INJECT STATISTICS e'[{"avg_size": 19, "columns": ["co\U0001F62Cl1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 32008954634, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 50000000000}, {"avg_size": 17, "columns": ["\\"cOl1_6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 15142948695, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 50000000000}, {"avg_size": 11, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 39924246268, "histo_buckets": [{"distinct_range": 0, "num_eq": 0, "num_range": 0, "upper_bound": ""}, {"distinct_range": 0, "num_eq": 0, "num_range": 0, "upper_bound": "\\u00129="}, {"distinct_range": 1948954741539257900, "num_eq": 5186852661064128369, "num_range": 1948954741539257932, "upper_bound": "$=yB\\u0019"}, {"distinct_range": 0, "num_eq": 500000000, "num_range": 90, "upper_bound": " X\\u0007|\\\\N"}, {"distinct_range": 0, "num_eq": 2000000000, "num_range": 9167716677200123861, "upper_bound": "\\u0011-IW1e\\u0004"}, {"distinct_range": 5225100196309358000, "num_eq": 3000000000, "num_range": 5225100196309357390, "upper_bound": "dQEQVsz\\u001a"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 50000000000}, {"avg_size": 14, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 47643813742, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 50000000000}, {"avg_size": 21, "columns": ["\\"\\ncol1 _1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 38985514806, "histo_buckets": [{"distinct_range": 0, "num_eq": 40000000, "num_range": 0, "upper_bound": ""}, {"distinct_range": 54.76735189330745, "num_eq": 3884233114840234851, "num_range": 100, "upper_bound": "\\u0019|W\'"}, {"distinct_range": 0, "num_eq": 4906494700740409147, "num_range": 4385500450337716104, "upper_bound": "I30\x7f\x7f4q\\u00066"}, {"distinct_range": 5.970714580953072, "num_eq": 0, "num_range": 40, "upper_bound": "|{ S1UH;<"}], "histo_col_type": "VARCHAR", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 50000000000}, {"avg_size": 19, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 36226772089, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 50000000000}, {"avg_size": 20, "columns": ["\\"col 1_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 10357113969, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 50000000000}, {"avg_size": 11, "columns": ["\\"co%qL1_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 7872431698, "histo_col_type": "", "name": "__auto__", "null_count": 13486338333, "row_count": 50000000000}]':::JSONB;
    rsg_test.go:765: CREATE TYPE rand_typ_0 AS ENUM ('aub', 'cshplz', 'upyvm', 'wvs');
    rsg_test.go:765: CREATE TYPE rand_typ_1 AS ENUM ('do', 'zmv', 'o');
    rsg_test.go:765: SET sql_safe_updates = false;;
    rsg_test.go:765: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:765: ALTER TABLE seed SPLIT AT VALUES (6769684676736133582);
    rsg_test.go:765: ALTER TABLE seed SPLIT AT VALUES (9176722205110967754);
    rsg_test.go:765: ALTER TABLE seed SPLIT AT VALUES (7091582256438051314);
    rsg_test.go:765: ALTER TABLE seed SPLIT AT VALUES (9179275154580411429);
    rsg_test.go:765: ALTER TABLE seed SPLIT AT VALUES (5364933277530799165);
    rsg_test.go:765: ALTER TABLE seed SPLIT AT VALUES (2787478851749190850);
    rsg_test.go:765: ALTER TABLE seed SPLIT AT VALUES (3937783166787413070);
    rsg_test.go:765: ALTER TABLE seed SCATTER;
    rsg_test.go:767: 
    rsg_test.go:768: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith1829937685
--- FAIL: TestRandomSyntaxSQLSmith (302.52s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #132728 sql/tests: TestRandomSyntaxSQLSmith failed [SIGSEGV in geos.Difference] [C-test-failure O-robot T-sql-queries branch-release-23.2]
- #120970 sql/tests: TestRandomSyntaxSQLSmith failed [SIGSEGV in geos.LineMerge] [A-spatial C-bug C-test-failure O-robot P-3 T-sql-queries branch-release-23.1.18-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-43508

--- a/pkg/sql/opt/memo/testdata/logprops/set
+++ b/pkg/sql/opt/memo/testdata/logprops/set
@@ -669,3 +669,74 @@ intersect-all
       ├── fd: (5)-->(6)
       ├── prune: (5,6)
       └── interesting orderings: (+5)
+
+# Regression test for #133221: if a column with an equivalency is remapped more
+# than once, create a new equivalency for each remapping.
+norm
+SELECT * FROM (SELECT x, x, COALESCE(NULL, x) FROM xy) INTERSECT VALUES (1, 2, 3);
+----
+intersect-all
+ ├── columns: x:9(int!null) x:10(int!null) coalesce:11(int!null)
+ ├── left columns: xy.x:1(int) xy.x:1(int) coalesce:5(int)
+ ├── right columns: column1:6(int) column2:7(int) column3:8(int)
+ ├── cardinality: [0 - 1]
+ ├── key: ()
+ ├── fd: ()-->(9-11), (11)==(9,10), (9)==(10,11), (10)==(9,11)
+ ├── interesting orderings: (+(9|10|11))
+ ├── project
+ │    ├── columns: coalesce:5(int!null) xy.x:1(int!null)
+ │    ├── key: (1)
+ │    ├── fd: (1)==(5), (5)==(1)
+ │    ├── prune: (1,5)
+ │    ├── interesting orderings: (+(1|5))
+ │    ├── scan xy
+ │    │    ├── columns: xy.x:1(int!null)
+ │    │    ├── key: (1)
+ │    │    ├── prune: (1)
+ │    │    └── interesting orderings: (+1)
+ │    └── projections
+ │         └── variable: xy.x:1 [as=coalesce:5, type=int, outer=(1)]
+ └── values
+      ├── columns: column1:6(int!null) column2:7(int!null) column3:8(int!null)
+      ├── cardinality: [1 - 1]
+      ├── key: ()
+      ├── fd: ()-->(6-8)
+      ├── prune: (6-8)
+      └── tuple [type=tuple{int, int, int}]
+           ├── const: 1 [type=int]
+           ├── const: 2 [type=int]
+           └── const: 3 [type=int]
+
+norm
+SELECT * FROM (SELECT x, x, COALESCE(NULL, x) FROM xy) EXCEPT VALUES (1, 2, 3);
+----
+except-all
+ ├── columns: x:9(int!null) x:10(int!null) coalesce:11(int!null)
+ ├── left columns: xy.x:1(int) xy.x:1(int) coalesce:5(int)
+ ├── right columns: column1:6(int) column2:7(int) column3:8(int)
+ ├── key: (10)
+ ├── fd: (11)==(9,10), (9)==(10,11), (10)==(9,11)
+ ├── interesting orderings: (+(9|10|11))
+ ├── project
+ │    ├── columns: coalesce:5(int!null) xy.x:1(int!null)
+ │    ├── key: (1)
+ │    ├── fd: (1)==(5), (5)==(1)
+ │    ├── prune: (1,5)
+ │    ├── interesting orderings: (+(1|5))
+ │    ├── scan xy
+ │    │    ├── columns: xy.x:1(int!null)
+ │    │    ├── key: (1)
+ │    │    ├── prune: (1)
+ │    │    └── interesting orderings: (+1)
+ │    └── projections
+ │         └── variable: xy.x:1 [as=coalesce:5, type=int, outer=(1)]
+ └── values
+      ├── columns: column1:6(int!null) column2:7(int!null) column3:8(int!null)
+      ├── cardinality: [1 - 1]
+      ├── key: ()
+      ├── fd: ()-->(6-8)
+      ├── prune: (6-8)
+      └── tuple [type=tuple{int, int, int}]
+           ├── const: 1 [type=int]
+           ├── const: 2 [type=int]
+           └── const: 3 [type=int]

--- a/pkg/sql/opt/memo/testdata/logprops/with
+++ b/pkg/sql/opt/memo/testdata/logprops/with
@@ -287,3 +287,16 @@ with &2 (cte)
       ├── prune: (10-12)
       └── cte-uses
            └── &2: count=1 used-columns=(3-5)
+
+# Regression test for #133221: if a column with an equivalency is remapped more
+# than once, create a new equivalency for each remapping.
+norm
+WITH foo AS (SELECT x, x, COALESCE(NULL, x) FROM xy) SELECT 1 FROM foo;
+----
+project
+ ├── columns: "?column?":9(int!null)
+ ├── fd: ()-->(9)
+ ├── prune: (9)
+ ├── scan xy
+ └── projections
+      └── const: 1 [as="?column?":9, type=int]

--- a/pkg/sql/opt/props/func_dep.go
+++ b/pkg/sql/opt/props/func_dep.go
@@ -546,21 +546,45 @@ func (f *FuncDepSet) CopyFrom(fdset *FuncDepSet) {
 }
 
 // RemapFrom copies the given FD into this FD, remapping column IDs according to
-// the from/to lists. Specifically, column from[i] is replaced with column
-// to[i] (see TranslateColSet).
-// Any columns not in the from list are removed from the FDs.
-func (f *FuncDepSet) RemapFrom(fdset *FuncDepSet, fromCols, toCols opt.ColList) {
+// the old/new lists. Specifically, column old[i] is replaced with column new[i]
+// (see TranslateColSet).
+//
+// Any columns not in the "old" list are removed from the FDs.
+func (f *FuncDepSet) RemapFrom(fdset *FuncDepSet, oldCols, newCols opt.ColList) {
 	f.CopyFrom(fdset)
 	colSet := f.ColSet()
-	fromSet := fromCols.ToSet()
+	fromSet := oldCols.ToSet()
 	if !colSet.SubsetOf(fromSet) {
 		f.ProjectCols(colSet.Intersection(fromSet))
 	}
-	for i := range f.deps {
-		f.deps[i].from = opt.TranslateColSetStrict(f.deps[i].from, fromCols, toCols)
-		f.deps[i].to = opt.TranslateColSetStrict(f.deps[i].to, fromCols, toCols)
+	n := 0
+	var newEquivCols []opt.ColSet
+	for i := 0; i < len(f.deps); i++ {
+		remappedFrom := opt.TranslateColSetStrict(f.deps[i].from, oldCols, newCols)
+		remappedTo := opt.TranslateColSetStrict(f.deps[i].to, oldCols, newCols)
+		if f.deps[i].equiv && remappedFrom.Len() != 1 {
+			// The original "from" column maps to more than one "to" column. Remove
+			// this FD, and keep track of the set of equivalent columns to be handled
+			// after the loop.
+			remappedFrom.UnionWith(remappedTo)
+			newEquivCols = append(newEquivCols, remappedFrom)
+			continue
+		}
+		f.deps[i].from = remappedFrom
+		f.deps[i].to = remappedTo
+		if n != i {
+			f.deps[n] = f.deps[i]
+		}
+		n++
 	}
-	f.key = opt.TranslateColSetStrict(f.key, fromCols, toCols)
+	f.deps = f.deps[:n]
+	f.key = opt.TranslateColSetStrict(f.key, oldCols, newCols)
+	if len(newEquivCols) > 0 {
+		for i := range newEquivCols {
+			f.addEquivalency(newEquivCols[i])
+		}
+		f.tryToReduceKey(opt.ColSet{} /* notNullCols */)
+	}
 }
 
 // ColsAreStrictKey returns true if the given columns contain a strict key for the

--- a/pkg/sql/opt/props/func_dep_test.go
+++ b/pkg/sql/opt/props/func_dep_test.go
@@ -1276,6 +1276,16 @@ func TestFuncDeps_RemapFrom(t *testing.T) {
 	to = opt.ColList{10, 30, 40, 50}
 	res.RemapFrom(abcde, from, to)
 	verifyFD(t, &res, "key(10); (10)-->(30,40,50)")
+
+	// Test a one-to-many relationship between remapped columns. An equality
+	// should be split into two, and the remapped key should be reduced.
+	eqFD := &props.FuncDepSet{}
+	eqFD.AddStrictKey(c(1, 2, 3), c(1, 2, 3))
+	eqFD.AddEquivalency(2, 3)
+	from = opt.ColList{1, 2, 2, 3}
+	to = opt.ColList{10, 30, 20, 50}
+	res.RemapFrom(eqFD, from, to)
+	verifyFD(t, &res, "key(10,50); (50)==(20,30), (20)==(30,50), (30)==(20,50)")
 }
 
 // Construct base table FD from figure 3.3, page 114:

